File: library/src/main/java/org/metalscraps/eso/lang/lib/config/AppConfig.java
package org.metalscraps.eso.lang.lib.config;

import java.nio.charset.Charset;
import java.nio.charset.StandardCharsets;
import java.util.regex.Pattern;

/**
 * Created by 안병길 on 2018-01-17.
 * Whya5448@gmail.com
 */
@SuppressWarnings("Annotator")
public class AppConfig {
	public static final Charset CHARSET = StandardCharsets.UTF_8;
	public static final Pattern POPattern = Pattern.compile("(#, fuzzy)?\\n?msgctxt \"([0-9-]+)()()()\"\\n*?msgid \"{1,2}?\\n?([\\s\\S]*?)\"\\n*?msgstr \"{1,2}?\\n?([\\s\\S]*?)\"\\n{2,}", Pattern.MULTILINE);
	public static final Pattern CSVPattern = Pattern.compile("\"()(([\\d]+?)-([\\d]+?)-([\\d]+?))\",\"([\\s\\S]*?)\",\"([\\s\\S]*?)\"\n", Pattern.MULTILINE);
	public static final Pattern CategoryConfig = Pattern.compile ("FileName:(.*)((\\r\\n)|(\\n))isDuplicate:(.*)((\\r\\n)|(\\n))type:(.*)((\\r\\n)|(\\n))indexLinkCount:(.*)((\\r\\n)|(\\n))index:(.*)((\\r\\n)|(\\n))", Pattern.MULTILINE );
	public static final String englishTitlePattern = "([渀-馤가-힣\\s]+)\\s?\\([\\w\\s]+\\)";
	public static final String ZANATA_DOMAIN = "http://www.dostream.com/zanata/";
}

File: library/src/main/kotlin/org/metalscraps/eso/lang/lib/config/AppConfig.kt
package org.metalscraps.eso.lang.lib.config

import org.metalscraps.eso.lang.lib.bean.Lang
import java.nio.charset.Charset
import java.nio.charset.StandardCharsets
import java.util.regex.Pattern

/**
 * Created by 안병길 on 2018-01-17.
 * Whya5448@gmail.com
 */


object AppConfig {
    val langFiles = arrayOf(
            Lang("kr.lang", writeFileName = false, beta = false),
            Lang("kb.lang", writeFileName = false, beta = true),
            Lang("tr.lang", writeFileName = true, beta = false)
    )

    val CHARSET: Charset = StandardCharsets.UTF_8
    val POPattern: Pattern = Pattern.compile("(#, fuzzy)?(\\r\\n|\\r|\\n)?msgctxt \"([0-9-]+)\"(\\r\\n|\\r|\\n)*?msgid \"{1,2}?(\\r\\n|\\r|\\n)?([\\s\\S]*?)\"(\\r\\n|\\r|\\n)*?msgstr \"{1,2}?(\\r\\n|\\r|\\n)?([\\s\\S]*?)\"(\\r\\n|\\r|\\n){2,}", Pattern.MULTILINE)
    val CategoryConfig: Pattern = Pattern.compile("FileName:(.*)((\\r\\n)|(\\n))isDuplicate:(.*)((\\r\\n)|(\\n))type:(.*)((\\r\\n)|(\\n))indexLinkCount:(.*)((\\r\\n)|(\\n))index:(.*)((\\r\\n)|(\\n))", Pattern.MULTILINE)
    val PATTERN_DESTINATION: Pattern = Pattern.compile("(\\[)(\\d+)(] = \\{\")(.+?)(\"},)")
    const val ZANATA_DOMAIN = "https://translate.zanata.org/"
}